apps/api/src/app/subscribers/e2e/mark-all-subscriber-messages.e2e.ts

```diff
@@ -0,0 +1,268 @@
+    } catch (error) {
+      expect(error.response.status).to.equal(404);
+      expect(error.response.data.message).to.equal(
+        `Subscriber ${fakeSubscriberId} does not exist in environment ${session.environment._id}, ` +
+          'please provide a valid subscriber identifier'
+      );
+    }
+  });
+
+  it('should mark all the subscriber messages as read', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    const notificationsFeedResponse = await getSubscriberNotifications(session, session.subscriberId);
+    expect(notificationsFeedResponse.totalCount).to.equal(5);
+
+    const messagesMarkedAsReadResponse = await markAllSubscriberMessagesAs(
+      session,
+      session.subscriberId,
+      MarkMessagesAsEnum.READ
+    );
+    expect(messagesMarkedAsReadResponse.data).to.equal(5);
+
+    const feed = await messageRepository.find({
+      _environmentId: session.environment._id,
+      subscriberId: session.subscriberId,
+      channel: ChannelTypeEnum.IN_APP,
+      seen: true,
+      read: true,
+    });
+
+    expect(feed.length).to.equal(5);
+    for (const message of feed) {
+      expect(message.seen).to.equal(true);
+      expect(message.read).to.equal(true);
+    }
+  });
+
+  it('should not mark all the messages as read if they are already read', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, session.subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    const notificationsFeedResponse = await getSubscriberNotifications(session, session.subscriberId);
+    expect(notificationsFeedResponse.totalCount).to.equal(5);
+
+    await messageRepository.update(
+      {
+        _environmentId: session.environment._id,
+        subscriberId: session.subscriberId,
+        channel: ChannelTypeEnum.IN_APP,
+        seen: false,
+        read: false,
+      },
+      { $set: { read: true, seen: true } }
+    );
+
+    const messagesMarkedAsReadResponse = await markAllSubscriberMessagesAs(
+      session,
+      session.subscriberId,
+      MarkMessagesAsEnum.READ
+    );
+    expect(messagesMarkedAsReadResponse.data).to.equal(0);
```

Comment:
this one will be enough to validate the test, won't it? or you wanted to revalidate the state just in case?